gateway-shell/src/main/java/org/apache/knox/gateway/shell/table/KnoxShellTable.java

```diff
@@ -86,6 +96,117 @@ public KnoxShellTable value(Comparable<? extends Object> value) {
     return col;
   }
 
+  public Conversions getConversion(Comparable<? extends Object> colIndex) {
+      Conversions type = null;
+
+      if (colIndex instanceof Double) {
+        type = Conversions.DOUBLE;
+      }
+      else if (colIndex instanceof Integer) {
+        type = Conversions.INTEGER;
+      }
+      else if (colIndex instanceof Float) {
+        type = Conversions.FLOAT;
+      }
+      else if (colIndex instanceof Byte) {
+        type = Conversions.BYTE;
+      }
+      else if (colIndex instanceof Short) {
+        type = Conversions.SHORT;
+      }
+      else if (colIndex instanceof Long) {
+        type = Conversions.LONG;
+      }
+      return type;
+    }
+
+    public double[] toDoubleArray(String colName) throws IllegalArgumentException {
+      List<Comparable<? extends Object>> col = values(colName);
+      double[] colArray = new double[col.size()];
+      Conversions conversionMethod = null;
+      for (int i = 0; i < col.size(); i++) {
+        if (i == 0) {
+          conversionMethod = getConversion(col.get(i));
```

Comment:
`getConversion` can return `null`? This will then fall through the switch statement and cause an empty double array to be returned?

Reply:
Changed that to throw an `IllegalArgumentException` in an else on line 120 should any type not specified in the enum be passed in to `getConversion.